core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,582 @@
+    await this.type(addVoiceArtistUserName, voiceArtists[2]);
+    await this.clickOn(voiceArtistSaveButton);
+  }
+
+  /**
+   * This function helps to choose notification type.
+   */
+  async optInToEmailNotifications(): Promise<void> {
+    await this.clickOn(feedbackToggleOff);
+  }
+
+  /**
+   * Exception function to verify the choice of receiving feedback
+   * and suggestion notifications via email
+   */
+  async expectEmailNotificationToBeActivated(): Promise<void> {
+    const input = await this.page.$('input[id="feedback-switch"]');
+
+    if (!input) {
+      throw new Error('Feedback switch input element not found.');
+    }
+    const isChecked = await input.evaluate(
+      (input) => (input as HTMLInputElement).checked);
+
+    if (isChecked) {
+      showMessage('suggestions notifications via email are enabled.');
+    } else {
+      showMessage('Feedback notifications via email are enabled.');
+    }
+  }
+
+  /**
+   * This funciton helps in deleting the exploration successfully.
+   */
+  async deleteExploration(): Promise<void> {
+    await this.clickOn(deleteExplorationButton);
+    await this.clickOn(deleteConfirmButton);
+  }
+
+  /**
+   * This function helps in verifying , if exploration is
+   * deleted successfully?
+   */
+  async expectExplorationToBeDeletedSuccessfullyFromCreatorDashboard(
+  ): Promise<void> {
+    await this.page.waitForTimeout(500);
+    try {
+      await this.page.goto(explorationUrlAfterPublished);
+      throw new Error('Exploration is not deleted successfully.');
+    } catch (error) {
+      showMessage('Exploration is Successfully deleted.');
+    }
+  }
+
+  /**
+   * This function helps in drafting the exploration.
+   */
+  async saveDraftExploration(): Promise<void> {
+    await this.clickOn(saveDraftButton);
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn('.e2e-test-save-draft-button');
+  }
+
+  /**
+   * This function checks whether changes has been drafted or not.
+   */
+  async expectExplorationToBeDraftedSuccessfully(): Promise<void> {
+    const button = await this.page.$('#tutorialSaveButton');
+
+    if (!button) {
+      throw new Error('Save button not found.');
+    }
+
+    const isDraftButtonDisabled = await button.evaluate(
+      (button) => (button as HTMLButtonElement).disabled);
+
+    if (isDraftButtonDisabled) {
+      showMessage('Changes have been successfully drafted.');
+    } else {
+      throw new Error('Failed to draft changes.');
+    }
+  }
+
+  /**
+   * This function helps in publishing the exploration
+   */
+  async publishExploration(): Promise<void> {
+    await this.saveDraftExploration();
+    await this.page.waitForTimeout(500);
+    await this.clickOn(publishButton);
+    await this.clickOn(publishConfirmButton);
+    await this.clickOn('.e2e-test-confirm-publish');
+    await this.clickOn(closePublishedPopUp);
+
+    await this.page.waitForTimeout(500);
```

Comment:
Instead of this, try waiting for some UI change that indicates that the exploration publishing was successful (pop-up, publish button becoming an unpublish button, etc.)